### ISSUE TITLE
Since vim 9.0.0910 vim-lsp popups jump to different splits

### ISSUE THREAD

**Issue by dotdash** [prabirshrestha/vim-lsp#1393]
### Steps to reproduce

    Have vim-lsp installed and an LSP that provides hover information.
    Open a file supported by that LSP
    Run `:LspHover --ui=float' on some item
    Run :split
    Run :LspHover --ui=float on some item in the new split

Instead of the popup appearing at the cursor location, the cursor jumps to the other split and the popup appears there.

### Expected behaviour

The popup from LspHover appears at the cursor position in both cases.

According to Bram, the change from vim 9.0.0910 simply uncovered a problem in vim-lsp and the problem should be fixed here. See https://github.com/vim/vim/issues/11594#issuecomment-1325111787

**Comment by nickspoons** [comment]
Another issue that appears to be related (related enough that I haven't opened a separate issue, but I can if that is preferred?):

When using `let g:lsp_diagnostics_float_cursor = 1` with a vertical split, leaving the cursor on a line containing an error in the right window results in the cursor moving to the left window, and the popup being displayed there.

```
vim -O emptybuffer.txt file/containing/errors
# Navigate cursor to error line in file/containing/errors
# Note that the cursor jumps to emptybuffer.txt
```

**Comment by nickspoons** [comment]
I managed to track down the window-changing for the issue I mentioned above to [autoload/lsp/internal/diagnostics/float.vim#L45](https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/internal/diagnostics/float.vim#L45): `call deletebufline(l:doc_win.get_bufnr(), 1, '$')
`. This is not a vim-lsp issue, it's a vim bug. I'll make some minimal repro steps and add them to the vim issue.

**Comment by nickspoons** [comment]
I thought I could repro but I can't now...

**Comment by nickspoons** [comment]
This was fixed in vim [9.0.0961](https://github.com/vim/vim/commit/3da8597fc07505c0f8839b0834aafe1c10ffb456)
